Realm/RLMUtil.mm

```diff
@@ -240,6 +241,44 @@ id RLMValidatedObjectForProperty(id obj, RLMProperty *prop, RLMSchema *schema) {
     return outArray;
 };
 
+NSArray *RLMCollectionValueForKey(NSString *key, RLMRealm *realm, RLMObjectSchema *objectSchema, size_t count, size_t (^indexGenerator)()) {
```

Comment:
the indexGenerator block should just take the index. index can be passed in from the loops in the following two methods rather than being incremented by each caller.